arch/Kconfig

```diff
@@ -46,6 +46,13 @@ config RISCV32
 	bool "RISCV32 architecture"
 	select HAS_DTS
 
+config SPARC
+	bool "SPARC architecture"
+	select BIG_ENDIAN
+	select USE_SWITCH_SUPPORTED
+	select USE_SWITCH
+	select ATOMIC_OPERATIONS_C
```

Comment:
why? shouldn't you be using GCC builtins?

Reply:
Current build doesn't link against libatomic and failed with the following error:
```
include/sys/atomic.h:223: undefined reference to `__atomic_fetch_or_4'
include/sys/atomic.h:270: undefined reference to `__atomic_fetch_and_4'
```

In fact, many arch currently selects pure C version of it.   Would it be OK with you if I say I'll create an issue to convert to builtins once this PR is merged?

Reply:
toolchain may not implement them, we had this issue with Nios II. fine to use C for now